Apsitvarkom.UnitTests/DataAccess/GoogleGeocoderTests.cs

```diff
@@ -0,0 +1,53 @@
+using System.Net;
+using System.Text;
+using Apsitvarkom.Configuration;
+using Apsitvarkom.DataAccess;
+using Moq;
+using Moq.Protected;
+
+namespace Apsitvarkom.UnitTests.DataAccess;
+
+public class GoogleGeocoderTests
+{
+    private static readonly IApiKeyProvider _apiKeyProvider = new Mock<IApiKeyProvider>().Object;
+
+    [Test]
+    public async Task SuccessfullyFindsFirstAddress()
+    {
+        const string title = "Hello World";
+        const string responseJsonString = "{\"results\":[{\"formatted_address\":\"" + title + "\"}, {\"formatted_address\": \"second address\"}]}";
+
+        var handlerMock = new Mock<HttpMessageHandler>();
+        var response = new HttpResponseMessage
+        {
+            StatusCode = HttpStatusCode.OK,
+            Content = new StringContent(responseJsonString),
+        };
+
+        handlerMock
+            .Protected()
```

Comment:
seeing this protected for the first time. damn 🤯

Reply:
yeah me too :)

Reply:
I haven't mocked an `HttpClient` before, so I think I followed this: https://dev.to/gautemeekolsen/how-to-test-httpclient-with-moq-in-c-2ldp